ogr/ogrsf_frmts/gpkg/ogrgeopackagedatasource.cpp

```diff
@@ -7723,7 +7723,56 @@ OGRLayer *GDALGeoPackageDataset::ExecuteSQL(const char *pszSQLCommand,
         SoftRollbackTransaction();
         return nullptr;
     }
-
+    else if (STARTS_WITH_CI(osSQLCommand, "SAVEPOINT"))
```

Comment:
As there's a bit of copy & paste related to that handling of transaction related SQL (including also the above lines for BEGIN/COMMIT/ROLLBACK) between the SQLite and GPKG driver, I'm wondering if this could be factored ?

Perhaps a  "bool OGRSQLiteBaseDataSource::ProcessTransactionSQL(const std::string& osSQLCommand)" method that would return true if it has fully processed the command ?